src/crawlers/browser_crawler.js

```diff
@@ -360,8 +360,43 @@ export default class BrowserCrawler extends BasicCrawler {
      * @internal
      */
     async _handleRequestFunction(crawlingContext) {
-        const { id } = crawlingContext;
-        const page = await this.browserPool.newPage({ id });
+        const newPageOptions = {
+            id: crawlingContext.id,
+        };
+
+        if (this.proxyConfiguration && this.launchContext.useIncognitoPages) {
+            const pageOptions = {};
+
+            const { session } = crawlingContext;
+            const { url } = this.proxyConfiguration.newProxyInfo(session && session.id);
+
+            const parsed = new URL(url);
+
+            // Playwright
+            pageOptions.proxy = {
+                server: parsed.origin,
+                username: parsed.username,
+                password: parsed.password,
+            };
+
+            // Puppeteer
+            pageOptions.proxyServer = parsed.origin;
+            pageOptions.proxyUsername = parsed.username;
+            pageOptions.proxyPassword = parsed.password;
```

Comment:
This would be better placed in the individual classes. Maybe add a `_getNewPageOptions` function in `BrowserCrawler` and then override it with `super` in `PuppeteerCrawler` and `PlaywrightCrawler`?